contracts/contracts/vault/VaultStorage.sol

```diff
@@ -148,8 +149,16 @@ contract VaultStorage is Initializable, Governable {
     uint256 constant MIN_UNIT_PRICE_DRIFT = 0.7e18;
     uint256 constant MAX_UNIT_PRICE_DRIFT = 1.3e18;
 
-    /// @notice Contract that swaps the vault's collateral assets
-    address public swapper = address(0);
+    /// @notice Collateral swap configuration.
+    /// @dev is packed into a single storage slot to save gas.
+    struct SwapConfig {
+        // Contract that swaps the vault's collateral assets
+        address swapper;
+        // Max allowed percentage the total value can drop below the total supply in basis points.
+        // For example 100 == 1%
+        uint16 allowedUndervalueBps;
```

Comment:
This is going to default to zero percent loss before it gets set, which is a good, safe number.